checks/http.go

```diff
@@ -63,31 +66,54 @@ func (c *HTTPChecker) Run(ctx *context.Context) pkg.Results {
 	return results
 }
 
-func (c *HTTPChecker) generateHTTPRequest(ctx *context.Context, check v1.HTTPCheck, connection *models.Connection) (*http.Request, error) {
-	client := http.NewClient().UserAgent("canary-checker/" + runner.Version)
+func CreateHTTPClient(ctx dutyctx.Context, auth types.Authentication) (*http.Client, error) {
+	client := http.NewClient()
 
-	for _, header := range check.Headers {
-		value, err := ctx.GetEnvValueFromCache(header, ctx.GetNamespace())
+	if !auth.Username.IsEmpty() {
+		user, err := ctx.GetEnvValueFromCache(auth.Username, ctx.GetNamespace())
 		if err != nil {
-			return nil, fmt.Errorf("failed getting header (%v): %w", header, err)
+			return nil, err
 		}
-
-		client.Header(header.Name, value)
+		password, err := ctx.GetEnvValueFromCache(auth.Password, ctx.GetNamespace())
+		if err != nil {
+			return nil, err
+		}
+		client.Auth(user, password)
+	} else if !auth.OAuth.IsEmpty() {
+		client.OAuth(middlewares.OauthConfig{
```

Comment:
I don't think this will work with client id / secret as env var - I don't think we need bearer / oauth for push connection, but if it is in there it should work. 

We should probably have a `duty/connections/CreateHttpClient(ctx, types.HttpConnection)`  that reuses code from: https://github.com/flanksource/mission-control/blob/main/connection/check.go  throw an error if the connection type doesn't support http clients)


```go
type HttpConnection struct {
      Authentication `yaml:",inline"`
       URL url `yaml:"url,omitempty"
      Bearer   EnvVar `yaml:"bearer,omitempty" json:"bearer,omitempty"`
      OAuth    OAuth  `yaml:"oauth,omitempty" json:"oauth,omitempty"`
      TLS TLSConfig  `yaml:"tls,omitempty" json:"tls,omitempty"`

func (http HttpConnection) Hydrate(ctx Context, namespace) (HttpConnection, error)

models.NewHttpConnection(ctx Context, models.Connection)  (HttpConnection, error)

models.NewConnectionFromHTTP(http HttpConnection)
```

`types.Authentication` probably should not have oauth / bearer as its not supported everywhere. 

We will probably need a similar model for AWS - if we want to add support for profiles / assumed roles / session names / web identities / etc.. - but that can definitely wait